### DISCUSSION TITLE
shm zero copy latency

### DISCUSSION THREAD

**Discussion by xixioba** [eclipse-zenoh/roadmap#50] [Q&A] [answered]
Is there an official shm (zero copy) transmission delay test? 

I tried to do some tests, but most of them are **a few hundred microseconds**. 
Compared with some other dds implementations , there seems to be some gaps(especially small data). 
ref https://github.com/ZhenshengLee/ros2_jetson_benchmarks

Am I missing something?

**Comment by Mallets** [comment, marked as answer]
Hello xixioba, 

Sorry for the late reply. Zenoh SHM support as of today is beneficial when sharing large payloads while for small payloads, as you noticed, it still has some gaps and we are aware of that. We are currently planning to improve the SHM layer to reduce latency across processes.

**Reply by xixioba** [reply]
Thank you for your reply. As a comparison, is there any hope of achieving the same performance as [iceoryx](https://iceoryx.io/v2.0.2/examples/iceperf/) in the plan?
In fact, I don't know exactly how iceoryx works, but it shows a very exciting latency performance

**Reply by Mallets** [reply]
That's certainly zenoh's goal!

**Reply by bryceschober** [reply]
Is supporting iceoryx as an SHM implementation a possibility?

**Reply by kydos** [reply]
That is a possibility. At the present time we have a native shared-memory implementation as we prefer to limit dependencies. Additionally the current design is distributed and avoids single daemons (a la IceOryx). It there any specific reason why you'd  like to have an integration with IceOryx?

**Reply by xixioba** [reply]
Do today's native shared-memory implementations have close and stable performance like [iceoryx](https://iceoryx.io/latest/examples/iceperf/)?
Maybe there’s hope in the future?I prefer zenoh’s distributed design to iceoryx!

**Reply by Mallets** [reply]
Hello @xixioba , we have recently added `ultra low latency support` in the latest zenoh 0.10.0-rc version: https://zenoh.io/blog/2023-10-03-zenoh-dragonite/#support-for-ultra-low-latency
